### ISSUE TITLE
how to test the shufflenet's classification model?

### ISSUE THREAD

**Issue by xiaocmxiao** [shicai/MobileNet-Caffe#38]
I use the classfy.py which caffe build-in,but I get the same result with different test pictures each time,Is anywhere wrong?the same to mobilenet's model.I find that the common thing in shufflenet and mobilenet,their last layer's type is convolution,I'd like to konw how to test this kind of models?thank you.

**Comment by shicai** [comment]
check your data processing steps.
the last conv is equivalent to fc layer. there is no difference.

**Comment by xiaocmxiao** [comment]
@shicai
I have trained the same data with googleNet and squeezeNet,the result is correct.So I think the data have no problem.After squeezeNet's Convolution,there are a ReLu and a Pooling Layer which is different with shuffleNet and mobileNet.Hence,I suspect the problem is in Convolution Layer.I want to know how to test
 it?Is any script?Of course,testing with (./build/tools/caffe test......) is ok!but I can't see the output of the class intuitively.thank you.

**Comment by shicai** [comment]
you can try `./build/tools/caffe` first, with this transform params:
```
  transform_param {
    scale: 0.017
    mirror: false
    crop_size: 224
    mean_value: [103.94,116.78,123.68]
  }
```

**Comment by xiaocmxiao** [comment]
@shicai
eh...
I just want to test a single image,but the classify.py returns an error value always(for example, I have cat,dog,fish,  it's return always is cat,although I input a fish or a dog !),but the googleNet's model(the same data trained) not have this problem.thank you.

**Comment by shicai** [comment]
you can not use single `classify.py` script for all kinds of models, since different models are trained in different ways. for this mobilenet model, you should change your script, and use ` scale: 0.017` param to evaluate, which means that `transformer.set_input_scale('data', 0.017)` should be added to your scripts.

**Comment by xiaocmxiao** [comment]
It's work!thank you.爱你么么哒

**Comment by dlyldxwl** [comment]
@shicai hello, you use scale: 0.017 and mean value: [103.94,116.78,123.68] param, and when i use the mobilenet  as a pretrain model, mean value of my data may not be the same as yours , i should use my data mean value in data layer when i fiuetune mobilenet model? thank you!

**Comment by shicai** [comment]
for natural RGB images, you can use my default setttings to finetune your model.

**Comment by dlyldxwl** [comment]
@shicai OK,I get it ,3q

**Comment by liuyiyiyiyi** [comment]
> It's work!thank you.爱你么么哒

你好，请问这个你是怎么解决的，我也是遇到了一样的情况，无论输入什么图片，输出的标签概率都是没有变化的

**Comment by xiaocmxiao** [comment]
@liuyiyiyiyi 训练的时候用了scale，测试的时候也必须用。你也可以训练的时候不用scale。
